app/text_buffer.py

```diff
@@ -732,15 +732,31 @@ def findCurrentPattern(self, direction):
       theRange = range(self.penRow)
     else:
       theRange = range(len(self.lines) - 1, self.penRow, -1)
-    for i in theRange:
-      found = localRe.search(self.lines[i])
+    if theRange:
```

Comment:
Not a big deal, but I wondered if lines 753 to 759 can be combined with 736 to 744 since they are similar. Maybe line 735 could be 

if not theRange:

and then theRange could be made to work with the 'for i in theRange'? I haven't carefully reviewed whether that would yield tighter code - so if it doesn't, then ignore this review comment. My gut is that it's likely to work out though.

Either way, thanks for address this. I'm looking forward to merging it.